src/types/services/GatewayService.ts

```diff
@@ -35,4 +37,5 @@ export interface PaymentGatwayAdapter {
   getBalance(request: GatewayGetGiftCardBalanceRequest): Promise<GatewayGetGiftCardBalanceResponse>
   validateAuthTransaction(request: GatewayInteraction): Promise<ValidateResponse>
   getAuthorizationIDKeyName(): Promise<AuthorizeIdKeyNameResponse>
+  service(request: SessionRequest): Promise<SessionResponse>
```

Comment:
should we just name this `session` instead of `service`?

Reply:
@kibo-kevinwatts Yes, this should be "session"